server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java

```diff
@@ -826,19 +827,34 @@ public TSFetchResultsResp fetchResults(TSFetchResultsReq req) {
       }
 
       QueryDataSet queryDataSet = queryId2DataSet.get(req.queryId);
-      TSQueryDataSet result = fillRpcReturnData(req.fetchSize, queryDataSet,
-          sessionIdUsernameMap.get(req.sessionId));
-
-      boolean hasResultSet = result.bufferForTime().limit() != 0;
-      if (!hasResultSet) {
-        queryId2DataSet.remove(req.queryId);
+      if (req.isAlign) {
+        TSQueryDataSet result = fillRpcReturnData(req.fetchSize, queryDataSet,
+            sessionIdUsernameMap.get(req.sessionId));
+        boolean hasResultSet = result.bufferForTime().limit() != 0;
+        if (!hasResultSet) {
+          queryId2DataSet.remove(req.queryId);
+        }
+        TSFetchResultsResp resp = getTSFetchResultsResp(getStatus(TSStatusCode.SUCCESS_STATUS,
+            "FetchResult successfully. Has more result: " + hasResultSet));
+        resp.setHasResultSet(hasResultSet);
+        resp.setQueryDataSet(result);
+        resp.setIsAlign(true);
+        return resp;
+      }
+      else {
+        TSQueryNonAlignDataSet nonAlignResult = fillRpcNonAlignReturnData(req.fetchSize, queryDataSet,
+            sessionIdUsernameMap.get(req.sessionId));
+        boolean hasResultSet = nonAlignResult.getTimeList().get(0).limit() != 0;
```

Comment:
I just want to confirm: Even if there is no result, will there still be an empty list in the TimeList? Otherwise, an IndexOutOfRangeException will be thrown.